packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,3167 @@
+  ///   style: MenuButton.styleFrom(foregroundColor: Colors.green),
+  ///   menuChildren: const <Widget>[ /* ... */ ],
+  ///   child: const Text('Button Label'),
+  /// ),
+  /// ```
+  static ButtonStyle styleFrom({
+    Color? foregroundColor,
+    Color? backgroundColor,
+    Color? disabledForegroundColor,
+    Color? disabledBackgroundColor,
+    Color? shadowColor,
+    Color? surfaceTintColor,
+    TextStyle? textStyle,
+    double? elevation,
+    EdgeInsetsGeometry? padding,
+    Size? minimumSize,
+    Size? fixedSize,
+    Size? maximumSize,
+    MouseCursor? enabledMouseCursor,
+    MouseCursor? disabledMouseCursor,
+    BorderSide? side,
+    OutlinedBorder? shape,
+    VisualDensity? visualDensity,
+    MaterialTapTargetSize? tapTargetSize,
+    Duration? animationDuration,
+    bool? enableFeedback,
+    AlignmentGeometry? alignment,
+    InteractiveInkFeatureFactory? splashFactory,
+  }) {
+    return TextButton.styleFrom(
+      foregroundColor: foregroundColor,
+      backgroundColor: backgroundColor,
+      disabledBackgroundColor: disabledBackgroundColor,
+      disabledForegroundColor: disabledForegroundColor,
+      shadowColor: shadowColor,
+      surfaceTintColor: surfaceTintColor,
+      textStyle: textStyle,
+      elevation: elevation,
+      padding: padding,
+      minimumSize: minimumSize,
+      fixedSize: fixedSize,
+      maximumSize: maximumSize,
+      enabledMouseCursor: enabledMouseCursor,
+      disabledMouseCursor: disabledMouseCursor,
+      side: side,
+      shape: shape,
+      visualDensity: visualDensity,
+      tapTargetSize: tapTargetSize,
+      animationDuration: animationDuration,
+      enableFeedback: enableFeedback,
+      alignment: alignment,
+      splashFactory: splashFactory,
+    );
+  }
+
+  @override
+  State<MenuButton> createState() => _MenuButtonState();
+
+  @override
+  List<DiagnosticsNode> debugDescribeChildren() {
+    return <DiagnosticsNode>[
+      ...menuChildren.map<DiagnosticsNode>((Widget child) {
+        return child.toDiagnosticsNode();
+      })
+    ];
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty<Widget>('leadingIcon', leadingIcon, defaultValue: null));
+    properties.add(DiagnosticsProperty<String>('child', child.toString()));
+    properties.add(DiagnosticsProperty<Widget>('trailingIcon', trailingIcon, defaultValue: null));
+    properties.add(DiagnosticsProperty<MenuStyle>('menuStyle', menuStyle, defaultValue: null));
+  }
+}
+
+class _MenuButtonState extends State<MenuButton> {
+  MenuHandle? _handle;
+  bool get _enabled => widget.menuChildren.isNotEmpty;
+  FocusNode? _internalFocusNode;
+  MenuController? _internalMenuController;
+  FocusNode get _buttonFocusNode => widget.focusNode ?? _internalFocusNode!;
+  bool _waitingToFocusMenu = false;
+  final GlobalKey _buttonKey = GlobalKey();
```

Comment:
Oh, I see. Makes sense. 

I noticed below that depending on how the menu is opened, different `context`s are used. Sometimes its the Builder context, sometimes its the GlobalKey context. Maybe we can unify that since I believe there is no meaning in those differences?

Reply:
I believe this little nit is still unaddressed:

> I noticed below that depending on how the menu is opened, different contexts are used. Sometimes its the Builder context, sometimes its the GlobalKey context. Maybe we can unify that since I believe there is no meaning in those differences?